specutils/spectra/spectrum_mixin.py

```diff
@@ -184,10 +182,19 @@ def velocity(self):
 
         return new_data
 
+    @deprecated('v1.13', alternative="with_spectral_axis_unit")
     def with_spectral_unit(self, unit, velocity_convention=None,
                            rest_value=None):
+        self.with_spectral_axis_unit(unit, velocity_convention=velocity_convention,
+                                     rest_value=rest_value)
+
+    def with_spectral_axis_unit(self, unit, velocity_convention=None,
+                           rest_value=None):
         """
-        Returns a new spectrum with a different spectral axis unit.
+        Returns a new spectrum with a different spectral axis unit. Note that this creates a new
+        object using the converted spectral axis and thus drops the original WCS, if it existed.
+        The original WCS will be stored in the ``original_wcs`` entry of the new object's ``meta``
+        dictionary.
```

Comment:
```suggestion
        object using the converted spectral axis and thus drops the original WCS, if it existed,
        replacing it with a :class:`~specutils.utils.wcs_utils.SpectralGWCS`. The original WCS
        will be stored in the ``original_wcs`` entry of the new object's ``meta`` dictionary.
```
That's about my only point left about the documentation, to make it clear that `new_spec.wcs` will be a different class of object. Maybe `SpectralGWCS` could also use a basic docstring, if only to link to `~gwcs.WCS`...